### PR TITLE
fix: updated code to allow for insecure, tls, mtls

### LINKED ISSUE #21: tests: we need to validate mTLS and TLS
Both the GRCP and HTTP clients will need to be able to run in TLS and mTLS configuration

### PR DESCRIPTION
**Pull request by z103cb**
Update client for MTLS and  TLS connections

Fixes #21

### CODE REVIEW

**Comment by dtrifiro** [comment]
It looks like CI was hanging for this branch, I canceled it manually.

File: src/caikit_nlp_client/grpc_client.py [hunk 1]
```diff
@@ -31,22 +33,31 @@ def make_channel(config: GrpcConfig) -> grpc.Channel:
 
     connection = f"{config.host}:{config.port}"
 
-    if config.insecure:
+    if not config.tls and not config.mtls:
```
Comment (dtrifiro) [inline src/caikit_nlp_client/grpc_client.py]:
```suggestion
    if (config.tls or config.mtls):
```

**Review by dtrifiro — APPROVED** [review]

File: src/caikit_nlp_client/grpc_client.py (hunk 1 above)
Comment (z103cb) [reply src/caikit_nlp_client/grpc_client.py]: Don't we need a `not` here ? 
```python
if not (config.tls or config.mlts):
```

Comment (z103cb) [reply src/caikit_nlp_client/grpc_client.py]: Fixed.

**Pull request merged**